core/src/apresolve.rs

```diff
@@ -56,35 +53,21 @@ fn select_ap(data: Vec<String>, fallback: &str, ap_port: Option<u16>) -> SocketA
     ap.unwrap_or_else(|| (String::from(fallback), port))
 }
 
-async fn try_apresolve(proxy: Option<&Url>) -> Result<ApResolveData, Box<dyn Error>> {
+async fn try_apresolve(http_client: &HttpClient) -> Result<ApResolveData, Box<dyn Error>> {
     let req = Request::builder()
         .method("GET")
         .uri(APRESOLVE_ENDPOINT)
         .body(Body::empty())
         .unwrap();
 
-    let response = if let Some(url) = proxy {
-        // Panic safety: all URLs are valid URIs
-        let uri = url.to_string().parse().unwrap();
-        let proxy = Proxy::new(Intercept::All, uri);
-        let connector = HttpConnector::new();
-        let proxy_connector = ProxyConnector::from_proxy_unsecured(connector, proxy);
-        Client::builder()
-            .build(proxy_connector)
-            .request(req)
-            .await?
-    } else {
-        Client::new().request(req).await?
-    };
-
-    let body = hyper::body::to_bytes(response.into_body()).await?;
+    let body = http_client.request_body(req).await?;
     let data: ApResolveData = serde_json::from_slice(body.as_ref())?;
 
     Ok(data)
 }
 
-pub async fn apresolve(proxy: Option<&Url>, ap_port: Option<u16>) -> AccessPoints {
-    let data = try_apresolve(proxy).await.unwrap_or_else(|e| {
+pub async fn apresolve(http_client: &HttpClient, ap_port: Option<u16>) -> AccessPoints {
```

Comment:
May be make this into a `ApResolver` `component!` so that Mercury, `Dealer` and `SpClient` can do `let access_point = self.session().apresolve(type)`.

Reply:
Done so in 15628842af2859eb11d6530431f2639b9c1c5868.

Reply:
Fixed in 3a7843d.